build-system/scripts/generate_circleci_config.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+# Operates on circleci (loaded as json) from stdin
+# Outputs filtered circleci without the jobs we don't need to run
+# NOTE: This uses the build manifest YAML file to filter the dependency graph in CircleCI BUT it is not one-to-one.
+# There is a heuristic here where we expect a job to be associated with a manifest job if it lists the build_manifest.yml job name in its command.
+import json
+import yaml
+from concurrent.futures import ProcessPoolExecutor, as_completed
+import subprocess
+
+# same functionality as query_manifest rebuildPatterns but in bulk
+def get_manifest_job_names():
+    manifest = yaml.safe_load(open("build_manifest.yml"))
+    return list(manifest)
+
+def has_associated_manifest_job(circleci_job, manifest_names):
+    steps = circleci_job.get("steps", [])
+    for step in steps:
+        run_info = step.get("run", {})
+        command = run_info.get("command", "")
+        for manifest_name in manifest_names:
+            if manifest_name in command:
+                return True
```

Comment:
I'd shoot for doing this more robust. `manifest_name in command` can lead to a lot of false positives, and small changes like rewriting the `run` command using the shorthand syntax (ie `run: echo foo` instead of `run: command: echo foo`) breaks this.

Perhaps we can search for the build commands we know (ie `build`, `cond_spot_run`, etc) on each run command (shorthand or not) and extract the argument with the job name to avoid false positives. Also, not sure if we should account for jobs that depend on more than one build-manifest project (possibly not).

And as I finished writing this I noticed that this is still in draft, so you had probably already considered all this and I just jumped the gun. Apologies if that's the case.

Reply:
Good suggestion

Reply:
That does seem to work, and I agree it's the less footgunny (plus you can search this string if confused and discover this script)